### PR TITLE
Converts cloud provider selection (intro step) to LitElement

### PR DESCRIPTION
**Pull request by mpmcroy**
This converts the cloud provider selection (`outline-intro-step`) to LitElement.

It's likely that this page will be updated as we add new cloud providers, so this PR converts the web component from Polymer 3 to LitElement.

Other minor updates: 
* Removes a redundant property.
  * Ideally I think we would pass the account model itself to the WC, instead of individual fields of the account.
* Makes CSS symmetrical.
* Moves event handling from `app-root` to `app`.

**Polymer 3 (right) vs LitElement (left)**
Screen Shot 2020-10-26 at 1 26 20 PMScreen Shot 2020-10-26 at 1 25 33 PM

**Polymer 3 (right) vs LitElement (left)**
Screen Shot 2020-10-26 at 11 48 24 AMScreen Shot 2020-10-26 at 1 24 56 PM

### CODE REVIEW

File: src/server_manager/web_app/ui_components/outline-intro-step.ts [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+    return html`
+      <outline-step-view>
+        <span slot="step-title">${this.localize('setup-title')}</span>
+        <span slot="step-description">${this.localize('setup-description')}</span>
+  
+        <div class="container">
+          <div id="digital-ocean" class="card" @tap="${this.connectToDigitalOceanTapped}">
+            <div class="card-header">
+              ${
+        this.digitalOceanEmail ? html`<div class="email">${this.digitalOceanEmail}</div>` :
```
Comment (fortuna) [inline src/server_manager/web_app/ui_components/outline-intro-step.ts:216]: This if-operator is really hard to read. Can you put that in a variable instead?
if (this.digitalOceanEmail) {
  cardHeader = ...
} else {
  cardHeader = ...
}

${cardHeader}

**Review by fortuna — APPROVED** [review]

File: src/server_manager/web_app/ui_components/outline-intro-step.ts (hunk 1 above)
Comment (mpmcroy) [reply src/server_manager/web_app/ui_components/outline-intro-step.ts]: That's a really good idea. Thanks.

I fixed some of the formatting as well.

**Comment by fortuna** [comment]
Let me know when I should take another look (and click on the review button)

**Comment by mpmcroy** [comment]
**UPDATE**

The RTL process isn't working the same way with LitElement web components as it did with Polymer. So the RTL friendly CSS we have isn't rendering as we would expect. There seem to be two issues:

1) The `dir` attribute is not being set on the LitElement web components. So the current `:host(:dir(rtl))` selectors won't select the right styling.

    I'm not entirely sure how this part of the codebase works. Initially I thought it was working because of the DirMixin module, but I noticed that it's not being applied to all elements (or even to the root element and then inherited by its children). We can probably write a small module that pulls the dir attribute from the root element.

    Screen Shot 2020-11-18 at 9 17 13 AM

1) The `:dir()` smart selector is not being applied.

    We can switch back to the `:host([dir=rtl])` selectors. Although they're not semantically the same, it should work for our use case since we always use the `:dir()` selector in conjunction with the `:host()` selector.
------------------------------------
Alternatively, the following works.

**Replace host selectors**
```
:host(:dir(rtl)) #aws .description ul {
  list-style-image: url("../images/check_orange_rtl.svg");
}
```
**With global selectors**
```
[dir=rtl] #aws .description ul {
  list-style-image: url("../images/check_orange_rtl.svg");
}
```

**Comment by mpmcroy** [comment]
Related to CSS RTL transcriber plugin #757

**Comment by mpmcroy** [comment]
Explicitly setting the `dir` attribute for LitElement web components. This, in addition to using the `:host([dir=rtl])` selectors, allows us to support RTL styling.

This seems like a great compromise until we find a replacement for the `DirMixin`. Thanks @JonathanDCohen for the suggestion.

**Comment by JonathanDCohen** [comment]
I wonder if the global dir selector works because LitElement applies the styling inside the shadow dom in some way that Polymer doesn't?

Anyways a small drive-by ask is to add to the README a section about LitElement specific gotchas like this until we can smooth everything out.  I'm sure Jon In The Future will forget about needing to manually add a dir tag even though it was my suggestion :P

**Review by fortuna — APPROVED** [review]
Setting the dir sounds reasonable. `:dir()` is only supported on Firefox, it won't work. Polymer "polyfills" it.

An alternative would be to do what the mixin does: if the dir attribute is not set, set it based on the html dir. Though if the html dir changes, the elements don't get updated. But we don't really need that.

A 3rd option would be to get the direction via `window.getComputedStyle(this).getPropertyValue("direction")`

**Pull request closed without merging**